### ISSUE TITLE
[expansion panel] JAWS does not read content of expansion panel when selected

### ISSUE THREAD

**Issue by jessithompson** [tyler-technologies-oss/forge#535]
**Describe the bug:**
With [JAWS sceenreader](https://www.freedomscientific.com/products/software/jaws/) active, selecting an expansion panel doesn't announce the content of the expansion panel. 

**To Reproduce:**
1. Go to `ButtonArea in Expansion Panel` story in the Forge Storybook (https://forge.tylerdev.io/main/?path=/story/components-button-area--in-expansion-panel)
2. Ensure JAWS is running
3. Navigate using keyboard to the expansion panel 

**Expected behavior:**
The screenreader should say something like "Heading toggle panel button collapsed. To activate press enter." (This is because the text in the expansion panel says "Heading".)

**Screenshots:**
![Screenshot 2024-04-18 at 3 43 33 PM](https://github.com/tyler-technologies-oss/forge/assets/31944890/07bae293-51e6-4420-a69c-6a2643462365)

**Please complete the following information:**
 - Forge version: v2.22.0
 - I have searched existing issues before creating this report? Y
 - Browser: Chrome, Firefox
 - Platform: desktop
 - OS: Windows

**Additional context:**
Today, the screenreader just says "toggle panel button collapsed. To activate press enter” when the expansion panel is selected.

**Comment by samrichardsontylertech** [comment]
This does seem confusing at first glance, but the accessible label of the button controlling the panel is actually "Toggle panel". This was done in the demo to give the button text descriptive of its action. The description from screen readers seems to do a pretty good job of that, though, so it should probably be changed to refer back to the visible content. When this is corrected you can expect JAWS to read "Heading button collapsed. To activate press enter."

Also I suspect that you might have navigated to the toggle button using the tab key (but I might be wrong so disregard this if you didn't). The visible part of the expansion panel heading isn't contained within a button so it isn't part of the tab order. Screen reader users will usually navigate between text with different shortcuts that allow them to read all text on the page. This is significant here because what you had in focus that JAWS read to you isn't the text of the heading, but a visually hidden button with its own label that proxies state styles out to the visible part of the panel. The text itself should read without issue to anyone viewing the page even without this change.

**Comment by jessithompson** [comment]
This is great to hear, I was using the tab button to navigate so I appreciate you addressing that. Thanks for the clear explanation!
